### ISSUE TITLE
Material Shadows not showing up when created dynamically

### ISSUE THREAD

**Issue by PikaChokeMe** [h6ah4i/android-materialshadowninepatch#11]
I have a vertical linear layout with a couple other vertical linear layouts hosted inside of it. I'm trying to use this code to create a little bit of space between each inner item as well as add a shadow to them, but no matter what I try I can't get shadows to appear when I create an object dynamically

My code is as follows

```
container.removeAllViews();

for (LinearLayout aLayout : itemsToAdd) {

    MaterialShadowContainerView msv = new MaterialShadowContainerView(getApplicationContext());
    msv.setShadowElevation(dp(6));
    msv.setShadowTranslationZ(dp(2));
    msv.setForceUseCompatShadow(true);

    ViewGroup.MarginLayoutParams p = new ViewGroup.MarginLayoutParams(ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    p.setMargins(dp(6), dp(6), dp(6), dp(45));

    msv.setLayoutParams(p);
    msv.addView(aLayout);
    container.addView(msv);

}
```

where container is a larger vertical linear layout that all of my smaller items are added to, and aLayout is each individual LinearLayout of an ArrayList<LinearLayout> collection. Right now I'm assuming there is something I've done wrong with the margins as no matter what I do I can't get them to show up at all. All I ever seem to get is one continuous block of white without a single shadow as shown below

![Nothing but White](http://i.imgur.com/QDghIcj.png)

Is there something wrong with the way I'm doing this, or is there an actual bug with this library?

**Comment by h6ah4i** [comment]
Thanks for your report. I'll take a look at this issue.

**Comment by PikaChokeMe** [comment]
Have you looked into this at all? I'm depending on this library for the shadows in app I'm trying to release very soon. Is this actually a bug or is my implementation incorrect?

**Comment by h6ah4i** [comment]
Sorry, I have not looked this at all yet. I guess this is a bug of the library and I'm going to fix it on this weekend.

**Comment by h6ah4i** [comment]
Hi. ~~I have confirmed that `MaterialShadowContainerView` can be used even if it created programmatically~~. (_EDIT: Sorry, I noticed that I was checking with the old version v0.6.2 of this library. The latest version v0.6.3 seems not working! I'll fix ASAP.._.) 

And also I have noticed an issue in your code. **Must apply margins to inner views** instead of the `MaterialShadowContainerView`s.

#### Tweaked code:

``` java
container.removeAllViews();

for (LinearLayout aLayout : itemsToAdd) {
    MaterialShadowContainerView.LayoutParams p1 = new MaterialShadowContainerView.LayoutParams(
        ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);
    p1.setMargins(dp(6), dp(6), dp(6), dp(45));

    aLayout.setLayoutParams(p1);

    MaterialShadowContainerView msv = new MaterialShadowContainerView(getApplicationContext());
    msv.setShadowElevation(dp(6));
    msv.setShadowTranslationZ(dp(2));
    msv.setForceUseCompatShadow(true);

    LinearLayout.LayoutParams p2 = new LinearLayout.LayoutParams(
        ViewGroup.LayoutParams.MATCH_PARENT, ViewGroup.LayoutParams.WRAP_CONTENT);

    msv.setLayoutParams(p2);
    msv.addView(aLayout);
    container.addView(msv);
}
```

**Comment by h6ah4i** [comment]
I have just published v0.6.4. 

The latest version has added a new constructor of `MaterialShadowContainerView` which has **four-parameters**. If you want to create a `MaterialShadowContainerView` instance programmatically, need to use the four-parameters constructor to specify the shadow-style resource ID explicitly.

``` java
MaterialShadowContainerView msv = new MaterialShadowContainerView(
        context, null, 0, R.style.ms9_DefaultShadowStyle);
```

Also, I have added a new [Add programmatically demo](https://github.com/h6ah4i/android-materialshadowninepatch/blob/master/example/src/main/java/com/h6ah4i/android/example/materialshadowninepatch/ProgrammaticallyAddDemoFragment.java), so please refer to its implementation for more detail of the usage.

Thanks!
